pyspecProcScripts/phasing.py

```diff
@@ -129,233 +149,219 @@ def ph1_real_Abs(s,dw,ph1_sel=0,ph2_sel=1,fl = None):
-    dt = np.diff(s.getaxis('t2')[:2]).item()
-    logging.info(strm("For hermitian test, my dwell time is",dt))
-    # am assuming the axis of the source data is not manipulated
-    data_for_peak = abs(s).mean_all_but(['t2'])
-    max_val = data_for_peak.data.max()
-    data_for_peak /= max_val
-    pairs = data_for_peak.contiguous(lambda x: abs(x) >
-            down_from_max)
-    peak_bounds = pairs[0,:] #gives the longest pair
-    peak_center = data_for_peak['t2':(0.1e-3,None)].argmax('t2').item()
-    s.setaxis('t2',lambda x: x-peak_center)
-    s.register_axis({'t2':0})
-    logging.debug(strm("closest to 0", s.getaxis('t2')[np.argmin(abs(s.getaxis('t2')-0))]))
-    max_shift = np.min(abs(peak_center-peak_bounds))
-    #}}}
-    #{{{construct test arrays for T2 decay and shift
-    if np.isscalar(rel_shift):
-        rel_shift = [-rel_shift,rel_shift]
-    shift_t = nddata(r_[rel_shift[0]:rel_shift[1]:shift_points*1j]*max_shift,'shift')
-    data_for_peak.setaxis('t2',lambda x:x-peak_center) #for plotting later
-    logging.debug(strm("closest to 0", s.getaxis('t2')[np.argmin(abs(s.getaxis('t2')-0))]))
-    #}}}
-    #{{{time shift and correct for T2 decay
-    orig_t2 = ndshape(s)['t2']
-    s.ft('t2',pad=orig_t2*2)
-    s *= np.exp(1j*2*pi*shift_t*
-            s.fromaxis('t2'))
-    s.set_units('shift','s')
-    s.ift('t2')
-    s = s['t2',:orig_t2]
-    logging.debug(strm("closest to 0", s.getaxis('t2')[np.argmin(abs(s.getaxis('t2')-0))]))
-    s.reorder(['t2'],first=False)
-    if fl:
-        fig_forlist, ax_list = plt.subplots(4,1, figsize=(10,10))
-        fl.next("hermitian diagnostic", fig=fig_forlist)
-        fig_forlist.suptitle(" ".join(["Hermitian Diagnostic"] + [j for j in [fl.basename] if j is not None]))
-        fl.image(s.C.mean_all_but(['shift','t2']), ax=ax_list[0])
-        ax_list[0].set_title('Shifted Data')
-    #}}}
-    #{{{make sure there's an odd number of points and set phase of center point to 0
-    logging.debug(strm("before taking the slice, endpoints are",s.getaxis('t2')[r_[0,-1]]))
-    logging.debug(strm("closest to 0", s.getaxis('t2')[np.argmin(abs(s.getaxis('t2')-0))]))
-    # {{{ this passes the assertion, but I
-    #     need jf_hack for the next step to
-    #     work -- this means the axis isn't
-    #     made of numerically exactly equal
-    #     steps, for some reason
-    jf_hack = True
-    test = np.diff(s.getaxis('t2'))
-    assert np.allclose(dt*np.ones_like(test),test)
-    if jf_hack:
-        x = s.getaxis('t2')
-        x[:] /= dt
-        x[:] = np.round(x)
-        x *= dt
+    orig_dt = s.get_ft_prop(direct, "dt")
+    s.ft(direct)
+    s = s[direct:frq_range]
+    s.ift(direct, pad=1024 * 16)
+    new_dt = s.get_ft_prop(direct, "dt")
+    non_aliased_range = r_[-aliasing_slop,aliasing_slop]*int(orig_dt/new_dt)
+    ini_delay = non_aliased_range[0]*new_dt
+    logger.debug(strm("ini delay is",ini_delay))
+    s = s[direct:(non_aliased_range[0],non_aliased_range[1])]
```

Comment:
indexing error but works now that I changed the format

Reply:
Hmm.  This isn’t right.  You’re treating it like these are time values, but they’re not, they are integer indices

Reply:
right but how you had it before you were indexing two different points instead of a range. This is now slicing based on the index values